devicetypes/plaidsystems/spruce-controller.src/spruce-controller.groovy

```diff
@@ -14,736 +10,525 @@
+			options: ["Zone 1", "Zone 2", "Zone 3", "Zone 4", "Zone 5", "Zone 6", "Zone 7", "Zone 8", "Zone 9", "Zone 10", "Zone 11", "Zone 12", "Zone 13", "Zone 14", "Zone 15", "Zone 16"]
+
+		//break for ease of reading settings
+		input title: "", description: "", displayDuringSetup: true, type: "paragraph", element: "paragraph"
+
+		//schedule specific settings
+		input title: "Schedule setup", displayDuringSetup: true, type: "paragraph", element: "paragraph",
+			description: "These settings only effect when the controller is switched to the on state."
+		input "splitCycle", "bool", title: "Cycle scheduled watering time to reduce runoff?", required: false, displayDuringSetup: true
+		input "valveDelay", "integer", title: "Delay between valves when a schedule runs? (seconds)", required: false, displayDuringSetup: true
+
+		input title: "Schedule times", displayDuringSetup: true, type: "paragraph", element: "paragraph",
+			description: "Set the minutes for each zone to water anytime the controller is switched on."
+		input name: "z1Duration", type: "integer", title: "Zone 1 schedule minutes"
+		input name: "z2Duration", type: "integer", title: "Zone 2 schedule minutes"
+		input name: "z3Duration", type: "integer", title: "Zone 3 schedule minutes"
+		input name: "z4Duration", type: "integer", title: "Zone 4 schedule minutes"
+		input name: "z5Duration", type: "integer", title: "Zone 5 schedule minutes"
+		input name: "z6Duration", type: "integer", title: "Zone 6 schedule minutes"
+		input name: "z7Duration", type: "integer", title: "Zone 7 schedule minutes"
+		input name: "z8Duration", type: "integer", title: "Zone 8 schedule minutes"
+		input name: "z9Duration", type: "integer", title: "Zone 9 schedule minutes"
+		input name: "z10Duration", type: "integer", title: "Zone 10 schedule minutes"
+		input name: "z11Duration", type: "integer", title: "Zone 11 schedule minutes"
+		input name: "z12Duration", type: "integer", title: "Zone 12 schedule minutes"
+		input name: "z13Duration", type: "integer", title: "Zone 13 schedule minutes"
+		input name: "z14Duration", type: "integer", title: "Zone 14 schedule minutes"
+		input name: "z15Duration", type: "integer", title: "Zone 15 schedule minutes"
+		input name: "z16Duration", type: "integer", title: "Zone 16 schedule minutes"
+
+		input title: "Version", description: VERSION, displayDuringSetup: true, type: "paragraph", element: "paragraph"
 	}
-    
-    preferences {
-    input description: 'If you have a rain sensor wired to the rain sensor input on the Spruce controller, turn it on here.', displayDuringSetup: true, type: 'paragraph', element: 'paragraph', title: 'Rain Sensor'
-    input description: 'The SYNC SETTINGS button must be pressed after making a change to the Rain sensor:', displayDuringSetup: false, type: 'paragraph', element: 'paragraph', title: ''     
-    input 'RainEnable', 'bool', title: 'Rain Sensor Attached?', required: false, displayDuringSetup: true
```

Comment:
Since this variable name was changed, is there any logic that will also honor the old name so that things don't break for old installations?

Reply:
I understand the concern, but I'm not sure this is an issue, since the old installations are already broken after the release of the new app.  The device is not useable in the new app since none of the valves are displayed and the smartapp is also no longer available in ST.

This updated dth no longer uses the smartapp and is automations friendly since it uses the child devices.  I think this makes it a better fit overall for the ST platform.